js/viewer/annotation/base.js

```diff
@@ -132,6 +110,18 @@ function Base(attrs){
     };
 }
 
+/**
+ * This functions check to see if the properties mentioned in this.addedProperties are present in the attribute list which is used to initialize this object.
+ * @param {object} attrs
+ */
+Base.prototype.updateAddedProperties = function (attrs) {
```

Comment:
You don't need this function, you can do it directly in the `setAttributesByJSON` without the need to iterate over the `attrs` again. So

```javascript
Base.prototype.setAttributesByJSON = function (attrs) {
  var attr,
        value,
        discardedAttributes = ["id", "style"];

  for (attr in attrs) {
    if (!attr || !attrs.hasOwnProperty(attr) || discaredAttributes.indexOf(attr) === -1) continue;
    value = attrs[attr];
    
    if (attr in this._addedProperties) {
      this._addedProperties[attr] = value;
    } else if (attr in this._attrs) {
      this._attrs[attr] = value;
    } else {
      this._ignoredAttributes[attr] = value;
    }
       
  
  }
};
```

Reply:
The reason I had this function was because, earlier I had the function `setAttributesBySVG` and I needed to update the `_addedProperties` in that too, so instead of doing the same thing twice, I created function for it.
The other reason was to emphasize on the need for that processing. Having a separate function provides that emphasis.

Should I remove it?

Reply:
If it's just thing one place then yes. Right now, you're iterating over the list of attributes twice. If you do it like I suggested you're going over the attribute list only once.

Reply:
Done